src/Sentry.Unity.Editor/AutoInstrumentation/ModuleReadWriter.cs

```diff
@@ -46,15 +132,65 @@ private static bool TryReadSymbols(this ModuleDefinition module)
             }
         }
 
-        public static void Write(StrongNameKeyPair? key, bool hasSymbols, ModuleDefinition module, string file)
+        internal TypeDefinition GetTypeDefinition(ModuleDefinition module, Type type)
         {
-            var parameters = new WriterParameters
+            var typeDefinition = module.GetType(type.FullName);
+            if (typeDefinition is null)
             {
-                WriteSymbols = hasSymbols,
-                StrongNameKeyPair = key
-            };
+                throw new ArgumentException($"Failed to get requested type definition in {module.Name}", type.FullName);
+            }
+
+            return typeDefinition;
+        }
+
+        internal MethodDefinition GetMethodDefinition(TypeDefinition typeDefinition, string name, Type[]? requiredParameters = null)
+        {
+            foreach (var method in typeDefinition.Methods)
+            {
+                if (method?.Name == name)
+                {
+                    switch (requiredParameters)
+                    {
+                        case null when !method.HasParameters:
+                            return method;
+                        case null:
+                            continue;
+                    }
+
+                    if (method.Parameters.Count != requiredParameters.Length)
+                    {
+                        continue;
+                    }
+
+                    // We go over all the required parameters and compare them to the parameters found in the method.
+                    var hasMatchingParameters = true;
+                    foreach (var parameter in requiredParameters)
+                    {
+                        var parameterDefinitions = method.Parameters.Where(p =>
+                            string.Equals(
+                                p.ParameterType.FullName,
+                                parameter.FullName,
+                                StringComparison.Ordinal))
+                            .ToList();
+
+                        if (parameterDefinitions.Count == 0)
+                        {
+                            hasMatchingParameters = false;
+                            break;
+                        }
+                    }
+
+                    if (hasMatchingParameters)
+                    {
+                        return method;
+                    }
+                }
+            }
 
-            module.Write(file, parameters);
+            throw new Exception(
+                $"Failed to find method '{name}' " +
+                $"in '{typeDefinition.FullName}' " +
+                $"with parameters: '{(requiredParameters is not null ? string.Join(",", requiredParameters.AsEnumerable()) : "none")}'");
         }
```

Comment:
Here's a cleaner version of the above method.

```csharp
internal MethodDefinition GetMethodDefinition(TypeDefinition typeDefinition, string name, Type[]? requiredParameters = null)
{
    requiredParameters ??= Array.Empty<Type>();

    var matchingMethod = typeDefinition.Methods
        .FirstOrDefault(method =>
            method.Name == name &&
            requiredParameters.Length == method.Parameters.Count &&
            
            // This is the equivalent to the current code
            // requiredParameters.All(parameter =>
            //     method.Parameters.Any(p => p.ParameterType.FullName == parameter.FullName))

            // I think you want this instead, as it will match in position
            method.Parameters.Select(p => p.ParameterType.FullName)
                .SequenceEqual(requiredParameters.Select(x => x.FullName))
        );

    if (matchingMethod == null)
    {
        throw new Exception(
            $"Failed to find method '{name}' " +
            $"in '{typeDefinition.FullName}' " +
            $"with parameters: '{string.Join<Type>(",", requiredParameters)}'");
    }

    return matchingMethod;
}
```